examples/colors/store.rs

```diff
@@ -0,0 +1,193 @@
+                        fields.next(),
+                        fields.next(),
+                        fields.next(),
+                        fields.next().and_then(|f| str::parse::<f32>(f).ok()),
+                        fields.next().and_then(|f| str::parse::<f32>(f).ok()),
+                        fields.next().and_then(|f| str::parse::<f32>(f).ok()),
+                    ) {
+                        let rgba = gdk::RGBA::new(red / 255.0, green / 255.0, blue / 255.0, 1.0);
+
+                        let pos = ((red as usize & 0xFF) << 16)
+                            | ((green as usize & 0xFF) << 8)
+                            | blue as usize;
+
+                        self.colors
+                            .borrow_mut()
+                            .get_mut(pos as usize)
+                            .map(|c| c.set_property("color", rgba));
+                        obj.items_changed(pos as u32, 0, 1);
+
+                        size += 1;
+                    }
+                }
+            }
+
+            self.parent_constructed(obj);
+        }
+
+        fn properties() -> &'static [ParamSpec] {
+            static PROPERTIES: Lazy<Vec<ParamSpec>> = Lazy::new(|| {
+                vec![
+                    ParamSpec::new_uint(
+                        "size",
+                        "size",
+                        "size",
+                        u32::MIN,
+                        u32::MAX,
+                        0,
+                        ParamFlags::READWRITE,
+                    ),
+                    ParamSpec::new_uint(
+                        "limit",
+                        "limit",
+                        "limit",
+                        u32::MIN,
+                        u32::MAX,
+                        0,
+                        ParamFlags::READWRITE,
+                    ),
+                ]
+            });
+            PROPERTIES.as_ref()
+        }
+
+        fn set_property(&self, _obj: &Self::Type, _id: usize, value: &Value, pspec: &ParamSpec) {
+            match pspec.name() {
+                "size" => self.set_size(_obj, value.get().unwrap()),
+                "limit" => {
+                    self.limit.replace(value.get().unwrap());
+                }
+                _ => unimplemented!(),
+            }
+        }
+
+        fn property(&self, _obj: &Self::Type, _id: usize, pspec: &ParamSpec) -> Value {
+            match pspec.name() {
+                "size" => (self.size.borrow()).to_value(),
+                "limit" => (self.limit.borrow()).to_value(),
+                _ => unimplemented!(),
+            }
+        }
+    }
+
+    impl ColorStore {
+        pub fn set_size(&self, _obj: &super::ColorStore, size: u32) {
+            let old_size = *self.size.borrow();
+
+            *self.size.borrow_mut() = size;
+
+            if size > old_size {
+                _obj.items_changed(old_size, 0, size - old_size);
```

Comment:
you prefix a param with _ when it's not used, except it's used so drop the `_` prefix :-)